### PR TITLE
Implement native conformance test runner

### PR DESCRIPTION
**Pull request by jcking**

### CODE REVIEW

File: conformance/BUILD.bazel [hunk 1]
```diff
@@ -24,47 +31,60 @@ ALL_TESTS = [
...
+
+go_test(
+    name = "go_default_test",
+    size = "small",
+    srcs = [
+        "conformance_test.go",
+    ],
+    tags = [
+        "manual",
+        "notap",
```
Comment (TristonianJones) [inline conformance/BUILD.bazel:60]: Should we put `notap` here?

File: conformance/conformance_test.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+	tests     testsFlag
+	skipTests skipTestsFlag
+
+	envWithMacros *cel.Env
+	envNoMacros   *cel.Env
+)
+
+func init() {
+	flag.BoolVar(&dashboard, "dashboard", false, "Dashboard.")
+	flag.Var(&tests, "tests", "Paths to run, separate by a comma.")
```
Comment (TristonianJones) [inline conformance/conformance_test.go:75]: Could you document an example path?

File: conformance/conformance_test.bzl [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    native.sh_test(
+        name = name,
+        size = "small",
+        srcs = ["//conformance:conformance_test.sh"],
+        args = ["$(location //conformance:go_default_test)"] + _conformance_test_args(data, skip_tests, dashboard),
+        data = ["//conformance:go_default_test"] + data,
+        tags = [
+            "guitar",
+            "manual",
+            "notap",
```
Comment (TristonianJones) [inline conformance/conformance_test.bzl:56]: We should probably nix references to guitar and notap here.

File: conformance/BUILD.bazel (hunk 1 above)
Comment (jcking) [reply conformance/BUILD.bazel:60]: Yes, this test target is not meant to be run directly. It is invoked via the conformance_test macros.

File: conformance/conformance_test.bzl (hunk 3 above)
Comment (jcking) [reply conformance/conformance_test.bzl:56]: Its easier to leave them, as adding them back in downstream will be tedious.

File: conformance/conformance_test.go (hunk 2 above)
Comment (jcking) [reply conformance/conformance_test.go:75]: This tool is never meant to be invoked directly, only through the macros. Bazel passes through the paths in a way that `github.com/bazelbuild/rules_go/go/runfiles` understands. Which is a little programmatic.

**Review by TristonianJones — APPROVED** [review]

**Pull request merged**